ddht/v5_1/alexandria/resource_queue.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any, AsyncIterator, Collection, Container, Hashable, Set, TypeVar
+
+from async_generator import asynccontextmanager
+import trio
+
+TResource = TypeVar("TResource", bound=Hashable)
+
+
+class RemoveResource(Exception):
+    pass
+
+
+class ResourceQueue(Container[TResource]):
```

Comment:
Future readers will appreciate a docstring here, it took me a few minutes to figure out what this class was going! This is kind of like a [`CapacityLimiter`](https://trio.readthedocs.io/en/v0.12.1/reference-core.html#trio.CapacityLimiter) except each token has a name?

Reply:
More like each token is an actual "thing" but yes. similar constructs.